rllib/algorithms/sac/sac_rl_module.py

```diff
@@ -0,0 +1,106 @@
+from typing import Type
+from ray.rllib.algorithms.sac.sac_catalog import SACCatalog
+
+# from ray.rllib.algorithms.sac.sac_learner import QF_PREDS
+# from ray.rllib.core.models.base import ACTOR, CRITIC
+from ray.rllib.core.models.specs.typing import SpecType
+from ray.rllib.core.rl_module.rl_module import RLModule
+from ray.rllib.core.rl_module.rl_module_with_target_networks_interface import (
+    RLModuleWithTargetNetworksInterface,
+)
+from ray.rllib.models.distributions import Distribution
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import ExperimentalAPI, override
+
+CRITIC_TARGET = "critic_target"
+QF_PREDS = "qf_preds"
+
+
+@ExperimentalAPI
+class SACRLModule(RLModule, RLModuleWithTargetNetworksInterface):
```

Comment:
Can we add a very detailed docstring for this class describing the different Module architectures (depending on the spaces/settings)?

Maybe even with some text graphix, e.g.
```
blabla this is the RLModule for SAC. It consists of ...

[obs] -> [pi-encoder] -> [pi-head] -> [action]

[obs] + [action] -> [q-encoder] -> [q-head] -> [q-value]
[obs] + [action] -> [twin q-encoder] -> [q-head] -> [q-value]

...

```

Reply:
Note to self: We should probably do this for all RLlib built-in RLModule classes.